pkgs/development/python-modules/nose-timer/default.nix

```diff
@@ -1,20 +1,39 @@
-{ buildPythonPackage, fetchPypi, lib, nose, }:
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, nose
+, mock
+, parameterized
+, pytestCheckHook
+, termcolor
+}:
 
 buildPythonPackage rec {
   pname = "nose-timer";
-  version = "1.0.0";
+  version = "1.0.1";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "09hwjwbczi06bfqgiylb2yxs5h88jdl26zi1fdqxdzvamrkksf2c";
+  src = fetchFromGitHub {
+    owner = "mahmoudimus";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0xsai2l5i1av62y9y0q63wy2zk27klmf2jizgghhxg2y8nfa8x3x";
   };
 
   propagatedBuildInputs = [ nose ];
 
+  checkInputs = [
+    mock
+    parameterized
+    pytestCheckHook
```

Comment:
While this does work the project does not use pytest for its tests. Please migrated to nose.

https://github.com/mahmoudimus/nose-timer/blob/9091924926c1e28d37b754730e96a4306614bbbe/tox.ini#L13-L18

Reply:
Using `nosetests` now.

Strangely enough this change requires `nose` to be part of `checkInputs` otherwise `nosetests` is not found even if `nose` is mentioned in `propagatedBuildInputs` already.

Reply:
> Strangely enough this change requires nose to be part of checkInputs otherwise nosetests is not found even if nose is mentioned in propagatedBuildInputs already.

Yes, that's because `strictDeps = true;` for Python packages. When this attribute is true, only native build inputs are added to `PATH`.  This feature is mostly there to reduce the chance of breaking cross-compilation. `checkInputs` are assumed (for now) to always be native; testing with cross-compilation would typically require an emulator anyway.

Reply:
Thanks for the clarification.